service/src/main/java/bio/terra/cbas/controllers/PublicApiController.java

```diff
@@ -1,16 +1,49 @@
 package bio.terra.cbas.controllers;
 
 import bio.terra.cbas.api.PublicApi;
+import bio.terra.cbas.config.CromwellServerConfiguration;
 import bio.terra.cbas.model.SystemStatus;
+import bio.terra.cbas.model.SystemStatusSystems;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 
 @Controller
 public class PublicApiController implements PublicApi {
 
+  private final CromwellServerConfiguration cromwellConfig;
+
+  @Autowired
+  public PublicApiController(CromwellServerConfiguration cromwellConfig) {
+    this.cromwellConfig = cromwellConfig;
+  }
+
   @Override
   public ResponseEntity<SystemStatus> getStatus() {
-    return new ResponseEntity<>(new SystemStatus().ok(true), HttpStatus.OK);
+
+    String result;
+    boolean isOk;
+    try {
+      URL url = new URL(this.cromwellConfig.healthUri());
+      URLConnection connection = url.openConnection();
+      connection.setConnectTimeout(5000);
+      var stream = connection.getInputStream();
+
+      result = new String(stream.readAllBytes(), StandardCharsets.UTF_8);
+      isOk = true;
```

Comment:
In my code I don't use an explicit timeout but if you do this, maybe you want to set read timeout too? Both have infinite timeout by default.

Anyway there's a minor bug with this version; the opened stream is never closed (although likely it will eventually be closed when GC'd). Using try-with-resources is how I'd make sure the stream is closed:

```
            try (var stream = connection.getInputStream()) {
                result = new String(stream.readAllBytes());
                isOk = true;
            }
```

Reply:
Ooh, thanks! Fixing now.